### DISCUSSION TITLE
Grouped dependency updates

### DISCUSSION THREAD

**Discussion by aloutfi** [community/community#15001] [Code Security]
Dependabot is a pretty great system. It provides a centralized medium through which dependency vulnerabilities can be detected, frees an organization's security team from needing to pay _as much_ attention to vulnerabilities, and helps expose developers to new functionality in the packages they are working in.

One of the tradeoffs of the system however is that it can become quite noisy and manual. A frequent workflow I have observed is that dependabot will open a number of pull requests, and they will sit in review for a while. Part of this might be due to the fact that when a dependabot PR is merged, the subsequent dependabot PRs need to be rebased. This takes time out of a reviewers schedule for a largely trivial, point and click operation.

The system works, but there is room for improvement in terms of the "Time to Merge" metric. Making a habit out of manually merging and rebasing dependabot PRs should not be a pre-requisite to ensuring the security of a software stack. Developers will lean toward writing one off scripts via github's API to automate the process and roll their own solutions, introducing a potentially dangerous variance to the process.

From GitHub's public roadmap, I see [#148: Dependabot: grouped updates](https://github.com/github/roadmap/issues/148) as an issue slated for future implementation. This would be an excellent addition to the system and provide much needed automation to the process.

**Comment by Fitzbert-Fitz** [comment]
@aloutfi Next to grouped updates you might also want to use [merge queue](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue) to avoid the repeated rebasing/merging issue
